mesheryctl/internal/cli/root/system/restart.go

```diff
@@ -0,0 +1,208 @@
+		stop = exec.Command("docker-compose", "-f", utils.DockerComposeFile, "rm", "-f")
+		stop.Stderr = os.Stderr
+
+		if err := stop.Run(); err != nil {
+			return errors.Wrap(err, utils.SystemError("failed to stop meshery"))
+		}
+
+		// Mesheryctl uses a docker volume for persistence. This volume should only be cleared when user wants
+		// to start from scratch with a fresh install.
+		// if err := exec.Command("docker", "volume", "prune", "-f").Run(); err != nil {
+		// 	log.Fatal("[ERROR] Please install docker-compose. The error message: \n", err)
+		// }
+
+		log.Info("Meshery is stopped.")
+
+		// Reset Meshery config file to default settings
+		if utils.ResetFlag {
+			err := resetMesheryConfig()
+			if err != nil {
+				return errors.Wrap(err, utils.SystemError("failed to reset meshery config"))
+			}
+		}
+
+		if _, err := os.Stat(utils.MesheryFolder); os.IsNotExist(err) {
+			if err := os.Mkdir(utils.MesheryFolder, 0777); err != nil {
+				return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to make %s directory", utils.MesheryFolder)))
+			}
+		}
+
+		if _, err := os.Stat(utils.DockerComposeFile); os.IsNotExist(err) {
+			if err := utils.DownloadFile(utils.DockerComposeFile, fileURL); err != nil {
+				return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to download %s file from %s", utils.DockerComposeFile, fileURL)))
+			}
+		}
+
+		//////// FLAGS
+		// Control whether to pull for new Meshery container images
+		if skipUpdateFlag {
+			log.Info("Skipping Meshery update...")
+		} else {
+			err := updateMesheryContainers()
+			if err != nil {
+				return errors.Wrap(err, utils.SystemError("failed to update meshery containers"))
+			}
+		}
+
+		// Reset Meshery config file to default settings
+		if utils.ResetFlag {
+			err := resetMesheryConfig()
+			if err != nil {
+				return errors.Wrap(err, utils.SystemError("failed to reset meshery config"))
+			}
+		}
+
+		log.Info("Starting Meshery again ...")
+		start := exec.Command("docker-compose", "-f", utils.DockerComposeFile, "up", "-d")
+		start.Stdout = os.Stdout
+		start.Stderr = os.Stderr
+
+		if err := start.Run(); err != nil {
+			return errors.Wrap(err, utils.SystemError("failed to run meshery server"))
+		}
+		checkFlag := 0 //flag to check
+
+		//connection to docker-client
+		cli, err := client.NewEnvClient()
+		if err != nil {
+			return errors.Wrap(err, utils.SystemError("failed to create new env client"))
+		}
+
+		containers, err := cli.ContainerList(context.Background(), types.ContainerListOptions{})
+		if err != nil {
+			return errors.Wrap(err, utils.SystemError("failed to fetch the list of containers"))
+		}
+
+		//check for container meshery_meshery_1 running status
+		for _, container := range containers {
+			if container.Names[0] == "/meshery_meshery_1" {
+				log.Info("Opening Meshery in your browser. If Meshery does not open, please point your browser to http://localhost:9081 to access Meshery.")
+
+				//check for os of host machine
```

Comment:
@subhamkrai Can be not write a separate function for checking the host machine and then call it here?
It would increase the code readability.

Reply:
Okay,, we can write 2 separate functions for stop and restart.